diff --git a/src/hw/msr/Msr_linux.cpp b/src/hw/msr/Msr_linux.cpp
index 0783ebc1..05351d01 100644
--- a/src/hw/msr/Msr_linux.cpp
+++ b/src/hw/msr/Msr_linux.cpp
@@ -57,7 +57,7 @@ public:

 xmrig::Msr::Msr() : d_ptr(new MsrPrivate())
 {
-    if (system("/sbin/modprobe msr allow_writes=on > /dev/null 2>&1") != 0) {
+    if (system("modprobe msr allow_writes=on > /dev/null 2>&1") != 0) {
         LOG_WARN("%s " YELLOW_BOLD("msr kernel module is not available"), Msr::tag());

         d_ptr->available = false;
